src/Algebra/Primality.agda

```diff
@@ -0,0 +1,55 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Definitions of irreducibility, primality, coprimality.
+------------------------------------------------------------------------
+
+-- You're unlikely to want to use this module directly. Instead you
+-- probably want to be importing the appropriate module from
+-- `Algebra.Properties.CancellativeCommutativeSemiring`.
+
+{-# OPTIONS --without-K --safe #-}
+
+open import Algebra.Core using (Op₂)
+import Algebra.Divisibility
+open import Data.Product using (_×_)
+open import Data.Sum using (_⊎_)
+open import Function using (flip)
+open import Level using (_⊔_)
+open import Relation.Binary using (Rel; Symmetric)
+open import Relation.Nullary using (¬_)
+open import Relation.Unary using (Pred)
+
+module Algebra.Primality
+  {a ℓ} {A : Set a} (_≈_ : Rel A ℓ) (_*_ : Op₂ A) (0# 1# : A)
+  where
+
+open import Algebra.Divisibility _≈_ _*_
+
+------------------------------------------------------------------------------
+-- Definitions
+
+private
+  infix 4 _≉_
+
+  _≉_ : Rel A _
+  x ≉ y = ¬ (x ≈ y)
+
+Irreducible : Pred A (a ⊔ ℓ)
+Irreducible p = p ∤ 1#  ×  (∀ {x y} → (p ≈ (x * y)) → x ∣ 1# ⊎ y ∣ 1#)
+
+Prime : Pred A (a ⊔ ℓ)
+Prime p = p ≉ 0#  ×  p ∤ 1#  ×  ∀ {x y} → p ∣ (x * y) → p ∣ x ⊎ p ∣ y
+
+-- * It is easy to prove        Prime ==>  Irreducible.
+-- * In a GCDSemiring it holds  Prime <==> Irreducible
+--   (for example, in ℕ, ℤ).
```

Comment:
I'm not sure these comments add anything. If it's easy to prove then it will be done so somewhere in the library. We also don't yet have a notion of `GCDSemiring` so we shouldn't have comments referring to it yet!

Reply:
I set now
```
-- * It will be proved  Prime ==> Irreducible.
-- * In many special cases it holds  Prime <==> Irreducible
--   (for example, in ℕ, ℤ).
```
I think this is helpful, because many people know what is Prime for ℕ and ℤ.
And they can see that there it is equivalent to Irreducible.
And they will doubt of why Irreducible is at all introduced.

If you insist, then I shall remove this comment.